tests/drivers/can/api/src/canfd.c

```diff
@@ -238,6 +238,45 @@ ZTEST(canfd, test_canfd_get_capabilities)
 			  "CAN FD loopback mode not supported");
 }
 
+/**
+ * @brief Test sending CAN FD frame with too big payload.
+ */
+ZTEST(canfd, test_send_fd_dlc_out_of_range)
```

Comment:
There is already negative test that checks error code returned by can_send() when CAN frame ID is invalid:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/can/api/src/classic.c#L799
Here, I add similar test. It checks that can_send() returns -EINVAL when CAN frame has invalid DLC value.
Test variant for CAN FD.